src/python/pants/jvm/target_types.py

```diff
@@ -46,6 +46,16 @@ class JvmArtifactVersionField(StringField):
     )
 
 
+class JvmArtifactUrlField(StringField):
+    alias = "url"
+    required = False
+    help = (
+        "A URL that points to the location of this artifact. If specified, Pants will not fetch this artifact "
```

Comment:
Drive-by comment, it's helpful to put `\n\n` in between each distinct idea so we have separate paragraphs. Maybe one here and then another with the `file` nuance.